src/components/CheckForm/FormLayout/FormSidebar.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { GrafanaTheme2 } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { Icon, useStyles2 } from '@grafana/ui';
+import { css, cx } from '@emotion/css';
+
+export function FormSidebar({
+  sections,
+  onSectionSelect,
+  activeIndex,
+}: {
+  sections: Array<{ label: string; hasErrors: boolean; required?: boolean }>;
+  onSectionSelect: (index: number) => void;
+  activeIndex: number;
+}) {
+  const styles = useStyles2(getStyles);
+  return (
+    <ol className={styles.container}>
+      {sections.map(({ label: sectionTitle, required, hasErrors }, sectionIndex) => {
+        const prefix = hasErrors ? (
+          <Icon name={`exclamation-triangle`} color={config.theme2.colors.error.main} />
+        ) : (
+          <span className={css({ width: '16px', display: 'inline-block' })}>{sectionIndex + 1}</span>
+        );
+        return (
+          <li
+            key={sectionTitle}
+            onClick={() => onSectionSelect(sectionIndex)}
+            className={cx(styles.listItem, { [styles.active]: activeIndex === sectionIndex })}
+          >
+            <div className={css({ cursor: 'pointer' })}>
+              {prefix}&nbsp;{sectionTitle}
+              {required && ' *'}
+            </div>
```

Comment:
The spacing and alignment looks a little off here. I'd drop the `&nbsp` character and flex the container with a gap and `align-items: center`. I think as a rule spacing should be handled by the theme as much as possible.